packages/lambdas/src/sqs-to-converter.ts

```diff
@@ -161,6 +162,47 @@ function postProcessSidechainFHIRBundle(
   console.log(`Bundle being sent to FHIR server: ${fhirBundleStr}`);
   return JSON.parse(fhirBundleStr);
 }
+function postProcessSidechainFHIRBundleStep2(bundle: Bundle): FHIRBundle {
+  const binaryIdToResource: { [key: string]: Binary } = {};
+  if (bundle.entry) {
+    // first, get all Binary resources into a map for quick lookup,
+    // and remove them from the Bundle - also, remove DocumentReference resources
+    bundle.entry = bundle.entry.filter(entry => {
+      let shouldKeep = true;
+      if (entry.resource?.resourceType === "Binary") {
+        shouldKeep = false;
+        if (entry.resource.id) {
+          binaryIdToResource[entry.resource.id] = entry.resource;
+        }
+      } else if (entry.resource?.resourceType === "DocumentReference") {
+        shouldKeep = false;
+      }
+      return shouldKeep;
+    });
+    // next, populate the DiagnosticReport resources with the content of the Binary
+    for (const entry of bundle.entry) {
+      if (entry.resource?.resourceType === "DiagnosticReport") {
+        if (entry.resource.presentedForm && entry.resource.presentedForm.length > 0) {
+          if (entry.resource.presentedForm[0]) {
```

Comment:
@leite08 in case you look at this - using `entry.resource.presentedForm[0]` was intentional, compiler was complaining about that element being null later in the code otherwise